addon/components/animated-popper.js

```diff
@@ -0,0 +1,197 @@
+import Component from '@ember/component';
+import { run } from '@ember/runloop';
+import { addObserver, removeObserver } from '@ember/object/observers';
+
+import { action } from 'ember-decorators/object';
+import { tagName } from 'ember-decorators/component';
+
+import { argument, type } from 'ember-argument-decorators';
+import { unionOf } from 'ember-argument-decorators/types';
+
+import { scheduler as raf, Token } from 'ember-raf-scheduler';
+
+import layout from '../templates/components/animated-popper';
+
+function hasTransition(element) {
```

Comment:
Just seemed like a pure enough function to extract, I like to keep simple functions separated. In the future this could be moved to `utils`, especially if needed in more than one place (`closest` for instance is actually pretty useful, I was considering pulling that into `ice-box`)